### ISSUE TITLE
Making `guzzle` dependencies optional for optional Remote Xdebug support

### ISSUE THREAD

**Issue by ek9** [leanphp/behat-code-coverage#15]
Currently for Remote XDebug functionality we are depending on Guzzle. As Guzzle is popular library, we have to be quite open about supported versions so this package does not conflict with versions of the library that developers can be using (related issues: #14 ).

It would be nice to gather feedback from everyone regarding what we do with Guzzle. I believe our options are:
- Move `guzzlehttp/guzle` from `required` to `suggested` in `composer.json`. This would make this dependency optional and only those, who use Remote XDebug would need to install it separately. This would allow developers to use any version of guzzle and ignore the Remote XDebug functionality if they do not use it. This would be reflected in the documentation etc.

- Leave the `guzzlehttp/guzzle` dependency as is.

I would like to request feedback from everyone: would moving from `required` to `suggested` break some kind of compatibility or workflow that you currently use? Are there any problems that you have encountered due to us depending on `guzzle` (i.e. if you use specific version of guzzle like in #10 )?

I personally would be in favour of move to `suggested`, but this might result in worse support for Remote Xdebug.

Thank you for your time and don't be shy to provide feedback!

**Comment by ek9** [comment]
I have released 2 new versions of this extension today:

    v3.1.0 - uses guzzlehttp/guzzle ~4.0||~5.0. This is legacy support release, mostly for those still stuck with 4.0 or 5.0 for whatever reason. This version will not likely to receive further updates, but can be used for transition period. It is temporarily maintained on v3.1 branch.
    v3.2.0 - uses guzzlehttp/guzzle ~6.0. This will be maintained and will receive any updates (master branch).

**Comment by thijswitteveen** [comment]
Thanks for updating!

Making it optional wouldn't break anything here, as I don't use the Remote XDebug functionality. If a clear errormessage is thrown in case Guzzle is missing it would be very obvious how to fix things. 

Perhaps providing adapters for different Guzzle versions is a solution for the longer term, as you can just maintain 1 version of this repo, which can be updated. Changes in Guzzle can be covered in the adapters.

**Comment by masterjus** [comment]
I'd prefer `suggestion` way. 
You can make some strategies which will be applied depending on project settings

**Comment by dannylewis-sheffield** [comment]
I agree with the `suggested` way, also update the docs and clean exit with info if running in in remote mode and `guzzle` is not available.
